File: api/src/main/java/dev/ulloasp/mlsuite/workspace/application/dto/WorkspaceContextDto.java
package dev.ulloasp.mlsuite.workspace.application.dto;

import java.util.List;

import dev.ulloasp.mlsuite.organization.application.dto.OrganizationDto;
import dev.ulloasp.mlsuite.organization.application.dto.OrganizationMembershipDto;
import dev.ulloasp.mlsuite.team.application.dto.TeamDto;

public record WorkspaceContextDto(
        WorkspaceUserDto user,
        List<OrganizationMembershipDto> memberships,
        List<OrganizationDto> organizations,
        OrganizationDto currentOrganization,
        OrganizationMembershipDto currentMembership,
        List<TeamDto> teams,
        WorkspacePermissionsDto permissions) {
}

File: frontend/src/api/workspace/dtos/workspace-context-dto.ts
/*
SPDX-License-Identifier: MIT
Copyright (c) 2025 Pablo Ulloa Santin
*/

import type { WorkspacePermissionsDto, OrganizationDto, OrganizationMembershipDto, TeamDto, InvitationDto, WorkspaceUserDto } from "./index";

export interface WorkspaceContextDto {
  user: WorkspaceUserDto;
  memberships: OrganizationMembershipDto[];
  organizations: OrganizationDto[];
  currentOrganization: OrganizationDto;
  currentMembership: OrganizationMembershipDto;
  teams: TeamDto[];
  invitations: InvitationDto[];
  permissions: WorkspacePermissionsDto;
}
